### PR TITLE
feat: soil interval editor proj settings

### PR DESCRIPTION
**Pull request by david-code**
## Description

Disables soil collection methods for intervals if they are disabled in the project level settings.

Should be reviewed after #678 is merged

### Checklist
- [x] Corresponding issue has been opened
- [ ] New tests added

### Related Issues
Part of #502

### CODE REVIEW

File: dev-client/package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
     "react-native-svg": "^14.0.0",
     "react-native-tab-view": "^3.5.2",
     "react-native-vector-icons": "^10.0.2",
-    "terraso-client-shared": "github:techmatters/terraso-client-shared#23fb61c",
+    "terraso-client-shared": "github:techmatters/terraso-client-shared#david-dev-branch",
```
Comment (paulschreiber) [inline dev-client/package.json:47]: issue: please use a description branch name

File: dev-client/src/screens/SoilScreen/SoilScreen.tsx [hunk 2]
```diff
@@ -27,7 +27,10 @@ import {EditIntervalModalContent} from 'terraso-mobile-client/screens/SoilScreen
 import {useMemo, useCallback} from 'react';
 import {
   LabelledDepthInterval,
+  ProjectSoilSettings,
```
Comment (paulschreiber) [inline dev-client/src/screens/SoilScreen/SoilScreen.tsx:30]: issue: fix unused variable warning.

File: dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx [hunk 3]
```diff
@@ -19,10 +19,12 @@ import {useDispatch, useSelector} from 'terraso-mobile-client/store';
 import {
   DepthInterval,
   sameDepth,
-  updateSoilDataDepthInterval,
+  updateSoilDataDepthIntervalAsync,
   soilPitMethods,
   methodEnabled,
+  methodRequired,
```
Comment (paulschreiber) [inline dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx:25]: issue: fix unused variable warning.

File: dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx [hunk 4]
```diff
@@ -84,13 +105,24 @@ export const EditIntervalModalContent = ({siteId, depthInterval}: Props) => {
...
+    (method: SoilPitMethod) => (newValue: boolean) => {
+      dispatch(
+        updateSoilDataDepthIntervalAsync({
+          siteId,
+          depthInterval,
+          [methodEnabled(method)]: newValue,
+        }),
+      );
+    },
+    [dispatch],
```
Comment (paulschreiber) [inline dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx:118]: issue: fix dependency array.

File: dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx [hunk 5]
```diff
@@ -181,3 +206,36 @@ const AddButton = ({action, isDisabled}: AddButtonProps) => {
...
+  value,
+  updateEnabled,
+}: SwitchProps) => {
+  const {t} = useTranslation();
+
+  const label = useMemo(
+    () =>
+      t(`soil.collection_method.${method}`) +
+      (isRequired ? ` (${t('general.required')})` : ''),
+    [method, isRequired],
```
Comment (paulschreiber) [inline dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx:229]: issue: fix depdendency.

File: dev-client/src/screens/Toasts.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import {useToast} from 'native-base';
```
Comment (paulschreiber) [inline dev-client/src/screens/Toasts.tsx:1]: issue: add copyright notice.

File: dev-client/src/screens/Toasts.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  useEffect(() => {
+    if (soilIdStatus === 'error') {
+      toast.show({
+        description: t('errors.soilId'),
+      });
+      dispatch(setSoilIdStatus('ready'));
+    }
+  }, [soilIdStatus, dispatch, toast, t]);
+
+  return <></>;
```
Comment (paulschreiber) [inline dev-client/src/screens/Toasts.tsx:22]: question: do you need to return anything? Can you return `</>` or `null`?

File: dev-client/src/translations/en.json [hunk 8]
```diff
@@ -316,7 +316,11 @@
       "data_portal_link_text": "LandPKS Data Portal",
       "privacy_policy_link_url": "https://landpotential.org/",
       "privacy_policy_link_text": "LandPKS Privacy Policy"
-    }
+    },
+    "required": "required"
+  },
+  "errors": {
+    "soilId": "Error saving soil ID info to backend."
```
Comment (paulschreiber) [inline dev-client/src/translations/en.json:323]:
```suggestion
    "soilId": "Error saving soil ID information."
```

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: dev-client/package.json (hunk 1 above)
Comment (david-code) [reply dev-client/package.json]: I don't understand this comment.

File: dev-client/src/screens/Toasts.tsx (hunk 7 above)
Comment (david-code) [reply dev-client/src/screens/Toasts.tsx:38]: I think you need to return some JSX for Typescript be happy. I could be wrong though.

File: dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx [hunk 9]
```diff
@@ -181,3 +205,36 @@ const AddButton = ({action, isDisabled}: AddButtonProps) => {
...
+  updateEnabled,
+}: SwitchProps) => {
+  const {t} = useTranslation();
+
+  const label = useMemo(
+    () =>
+      t(`soil.collection_method.${method}`) +
+      (isRequired ? ` (${t('general.required')})` : ''),
+    [t, method, isRequired],
+  );
```
Comment (paulschreiber) [inline dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx:229]: issue: don't concatenate strings. This won't work for RTL (and perhaps some other languages).  Perhaps something like 
```
(isRequired ? t('general.required_item_label', { item: t(`soil.collection_method.${method}`) }
            : t(`soil.collection_method.${method}`)
```

Where `required_item_label` is `{item}: (required)`.

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx (hunk 9 above)
Comment (paulschreiber) [reply dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx:229]: In some languages (RTL or not), the `required` string might need to appear in a different position in the string, i.e. `(required) blah`.  We can't assume word order is consistent.

Comment (david-code) [reply dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx]: Got it

**Comment by david-code** [comment]
@paulschreiber rebased with main and it seems most of this PR already made it's way to main. I updated the issue with the translation string nevertheless.

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**